src/scib_metrics/utils/_silhouette.py

```diff
@@ -43,21 +58,34 @@ def _silhouette_reduce(
     # clust_dists = jax.lax.fori_loop(
     #     0, D_chunk_len, lambda i, _data: _bincount(i, _data), (clust_dists, D_chunk, labels, label_freqs)
     # )[0]
-
     clust_dists = jax.vmap(partial(jnp.bincount, length=label_freqs.shape[0]), in_axes=(None, 0))(labels, D_chunk)
 
     # intra_index selects intra-cluster distances within clust_dists
     intra_index = (jnp.arange(D_chunk_len), jax.lax.dynamic_slice(labels, (start,), (D_chunk_len,)))
     # intra_clust_dists are averaged over cluster size outside this function
     intra_clust_dists = clust_dists[intra_index]
-    # of the remaining distances we normalise and extract the minimum
-    clust_dists = clust_dists.at[intra_index].set(jnp.inf)
-    clust_dists /= label_freqs
-    inter_clust_dists = clust_dists.min(axis=1)
+
+    if between_cluster_distances == "furthest":
+        # of the remaining distances we normalise and extract the maximum
+        clust_dists = clust_dists.at[intra_index].set(-jnp.inf)
+        clust_dists /= label_freqs
+        inter_clust_dists = clust_dists.max(axis=1)
+    elif between_cluster_distances == "mean_other":
+        clust_dists = clust_dists.at[intra_index].set(jnp.nan)
+        total_other_dists = jnp.nansum(clust_dists, axis=1)
+        total_other_count = jnp.sum(label_freqs) - label_freqs[jax.lax.dynamic_slice(labels, (start,), (D_chunk_len,))]
+        inter_clust_dists = total_other_dists / total_other_count
+    else:
```

Comment:
Would be great to check all options explicitly and have the else raise an error

Reply:
Thanks for the suggestion, I will implement it and update the pull request.